sway/commands/input.c

```diff
@@ -39,6 +39,30 @@ static struct cmd_handler input_config_handlers[] = {
 	{ "xkb_numlock", input_cmd_xkb_numlock },
 };
 
+/**
+ * Re-translate keysyms if a change in the input config could affect them.
+ */
+static void retranslate_keysyms(struct input_config *input_config) {
+	bool matched = false;
+	for (int i = 0; i < config->input_configs->length; ++i) {
+		struct input_config *ic = config->input_configs->items[i];
+		matched |= ic->identifier == input_config->identifier;
+
+		// the first configured xkb_layout
+		if (ic->xkb_layout) {
```

Comment:
Why not use `input_config->xkb_layout` directly?

Reply:
For some reason, I supposed that `xkb_layout` could be removed from an input configuration. This does not seem to be true, so now keysyms are re-translated if the changed input is the first one with a configured `xkb_layout`.

Reply:
Bump

Reply:
It is possible to have several configured inputs, each with a different `xkb_layout`. We need to check if `input_config` is the first one in the list, otherwise, it should not affect the translated bindings.